test/e2e/network/traffic_policy.go

```diff
@@ -0,0 +1,453 @@
+	portNum := 8080
+	specName := "traffic-policy"
+	tr := trafficPolicyTestResources{
+		zonesByPod:        make(map[string]string),
+		nodesByPod:        make(map[string]string),
+		clusterIPEndpoint: "",
+		endpointZone:      "",
+		endpointNode:      nil,
+		nodeportEndpoint:  "",
+		peerNode:          nil,
+		externalNode:      nil,
+	}
+	thLabels := map[string]string{labelKey: specName}
+	img := imageutils.GetE2EImage(imageutils.Agnhost)
+	ports := []v1.ContainerPort{{ContainerPort: int32(portNum)}}
+	dsConf := e2edaemonset.NewDaemonSet(specName+"-ds", img, thLabels, nil, nil, ports, "netexec")
+	ds, err := c.AppsV1().DaemonSets(f.Namespace.Name).Create(context.TODO(), dsConf, metav1.CreateOptions{})
+	framework.ExpectNoError(err, "error creating DaemonSet")
+
+	topologyHintsValue := "Disabled"
+	if topologyHints {
+		topologyHintsValue = "Auto"
+	}
+	svc := createServiceReportErr(c, f.Namespace.Name, &v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: specName,
+			Annotations: map[string]string{
+				v1.AnnotationTopologyAwareHints: topologyHintsValue,
+			},
+		},
+		Spec: v1.ServiceSpec{
+			Type:                  v1.ServiceTypeNodePort,
+			Selector:              thLabels,
+			ExternalTrafficPolicy: etp,
+			InternalTrafficPolicy: &itp,
+			Ports: []v1.ServicePort{{
+				Name:       "agnhost",
+				Port:       80,
+				TargetPort: intstr.FromInt(portNum),
+				Protocol:   v1.ProtocolTCP,
+			}},
+		},
+	})
+
+	err = wait.Poll(5*time.Second, framework.PodStartTimeout, func() (bool, error) {
+		return e2edaemonset.CheckRunningOnAllNodes(f, ds)
+	})
+	framework.ExpectNoError(err, "timed out waiting for DaemonSets to be ready")
+
+	// All Nodes should have same allocatable CPUs. If not, then skip the test.
+	schedulableNodes := map[string]*v1.Node{}
+	for _, nodeName := range e2edaemonset.SchedulableNodes(c, ds) {
+		schedulableNodes[nodeName] = nil
+	}
+
+	nodeList, err := c.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
+	framework.ExpectNoError(err, "Error when listing all Nodes")
+	var lastNodeCPU resource.Quantity
+	firstNode := true
+	for i := range nodeList.Items {
+		node := nodeList.Items[i]
+		if _, ok := schedulableNodes[node.Name]; !ok {
+			continue
+		}
+		schedulableNodes[node.Name] = &node
+
+		nodeCPU, found := node.Status.Allocatable[v1.ResourceCPU]
+		if !found {
+			framework.Failf("Error when getting allocatable CPU of Node '%s'", node.Name)
+		}
+		if firstNode {
+			lastNodeCPU = nodeCPU
+			firstNode = false
+		} else if !nodeCPU.Equal(lastNodeCPU) {
+			e2eskipper.Skipf("Expected Nodes to have equivalent allocatable CPUs, but Node '%s' is different from the previous one. %d not equals %d",
+				node.Name, nodeCPU.Value(), lastNodeCPU.Value())
+		}
+	}
+
+	framework.Logf("Waiting for %d endpoints to be tracked in EndpointSlices", len(schedulableNodes))
+
+	var finalSlices []discoveryv1.EndpointSlice
+	err = wait.Poll(5*time.Second, 3*time.Minute, func() (bool, error) {
```

Comment:
there are some constants defining this timeouts, check in the other files, I think that is services.go

Reply:
switched to svcReadyTimeout, resolving.